homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,394 @@
+"""WiZ Light integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from pywizlight import PilotBuilder, wizlight
+from pywizlight.bulblibrary import BulbType
+from pywizlight.exceptions import (
+    WizLightConnectionError,
+    WizLightNotKnownBulb,
+    WizLightTimeOutError,
+)
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP,
+    ATTR_EFFECT,
+    ATTR_HS_COLOR,
+    ATTR_RGB_COLOR,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    SUPPORT_COLOR_TEMP,
+    SUPPORT_EFFECT,
+    LightEntity,
+)
+from homeassistant.const import CONF_HOST, CONF_NAME
+import homeassistant.util.color as color_utils
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FEATURES_RGB = (
+    SUPPORT_BRIGHTNESS | SUPPORT_COLOR | SUPPORT_COLOR_TEMP | SUPPORT_EFFECT
+)
+SUPPORT_FEATURES_DIM = SUPPORT_BRIGHTNESS
+SUPPORT_FEATURES_WHITE = SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP
+
+
+# set poll interval to 15 sec because of changes from external to the bulb
+SCAN_INTERVAL = timedelta(seconds=15)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the WiZ Light platform from legacy config."""
+    # Assign configuration variables.
+    # The configuration check takes care they are present.
+    ip_address = config[CONF_HOST]
+    try:
+        bulb = wizlight(ip_address)
+    except WizLightConnectionError:
+        _LOGGER.error("Can't add bulb with ip %s", ip_address)
+        return False
+
+    # Add devices
+    async_add_entities([WizBulb(bulb, config[CONF_NAME])], update_before_add=True)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the WiZ Light platform from config_flow."""
+    # Assign configuration variables.
+    wizbulb = hass.data[DOMAIN][entry.entry_id]
+    # Add devices with defined name
+    async_add_entities([wizbulb], update_before_add=True)
+    return True
+
+
+class WizBulb(LightEntity):
```

Comment:
```suggestion
class WizBulbEntity(LightEntity):
```

Reply:
Just realized this is an entity.  We shouldn't use it in `async_setup_entry`.  

We can solve that by calling `bulb.getMac()` and `bulb.get_bulbtype()` there instead and save the result, and pass it into the entity when we create it in `light.py`

Reply:
I put a suggestion in `async_setup_entry` on how to accomplish that